### ISSUE TITLE
[DataGrid] Click on Expand icon of Master Detail also select the row checkbox 

### ISSUE THREAD

**Issue by Anand24816** [mui/mui-x#3945]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

When MasterDetail behavior is configured with rows with checkbox (`checkboxSelection` enabled), then expanding/collpasing the master panels also selects unselects the row.

Sample Code Sandbox
https://codesandbox.io/s/basicdetailpanels-material-demo-forked-7z3yn?file=/demo.tsx:5088-5105

### Expected behavior 🤔

Master Expand/collapse should not cause row selection.

### Steps to reproduce 🕹

Steps:

1. Enable the latest MasterDetail behvior by providing getDetailPanelContent
2. Enable the checkbox selection by adding `checkboxSelection`
3. Expand and collapse row with checkbox also getting affected.
4.

### Context 🔦

We want checkbox selection with master detail grid.

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

### Order ID 💳 (optional)

28914

**Comment by m4theushw** [comment]
Fixed by #3929

**Comment by AndreSilva1993** [comment]
@m4theushw what would be the advised way if I wish to get this behaviour though? It seems that the solution was to just stop the `onRowClick` from being invoked when click on interactive cells. However, I'm at the opposite side of things where I would like to select a row when it is expanded but seem to be unable to do so now.
